compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -728,9 +728,12 @@ class ROCMTargetBackend final : public TargetBackend {
         std::string header =
             llvm::formatv(R"TXT(
 ; To reproduce the .optimized.ll from the .linked.ll, run:
-; opt -S -mtriple={} -mcpu={} --passes='{}'
+; opt -S -mtriple={0} -mcpu={1} --passes='{2}' <.linked.ll>
 ; The flag '-S' to emit LLVMIR.
 ; The behavior of some passes depends on '-mtriple' and '-mcpu'
+;
+; To reproduce the .rocmasm from .optimized.ll, run:
+; llc -mtriple={0} -mcpu={1} <.optimized.ll> -o <out.rocmasm>
```

Comment:
You're missing either a pass pipeline or a -O

Reply:
Ideally I'd like to match the output from IREE. I don't think we add any passes or specify the pipelines on the API side either: https://github.com/iree-org/iree/blob/26f1a289d781f91b3d10444202411e7bccb2e201/compiler/plugins/target/ROCM/ROCMTarget.cpp#L240-L254

Reply:
... We definitely set the codegen opt level to either O2 or O3 (I think O3 is "Aggressive") though

It'll be near the TargetMachine creation

Reply:
O2 is the default:
```
Optimization level. [-O0, -O1, -O2, or -O3] (default = '-O2') 
```